Frontend/src/components/TaskList.tsx

```diff
@@ -0,0 +1,61 @@
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+import CsrfToken from './CsrfToken';
+
+export interface ITask {
+  id: number;
+  title: string;
+  status: boolean;
+}
+
+export async function getTasks(): Promise<ITask[]> {
+  const resp = await axios.get('list_tasks');
+  return resp.data.tasks;
+}
+
+export default function TaskList() {
+  const [tasks, setTasks] = useState<ITask[]>([]);
+
+  useEffect(() => {
+    const getResp = async () => {
+      const resp = await getTasks();
+      setTasks(resp);
+    };
+    getResp();
+  }, []);
+
+  CsrfToken();
+
+  return (
+    <div>
+      <div>
+        <h1>Pending</h1>
+        <ul>
+          {tasks
+            .filter((task) => task.status === false)
+            .map((task) => {
+              return (
+                <li key={task.id}>
+                  `{task.id}) {task.title}`
+                </li>
+              );
+            })}
+        </ul>
+      </div>
+      <div>
+        <h1>Completed</h1>
+        <ul>
+          {tasks
+            .filter((task) => task.status === true)
+            .map((task) => {
+              return (
+                <li key={task.id}>
+                  `{task.id}) {task.title}`
+                </li>
+              );
+            })}
```

Comment:
I have an extra challenge for you :)
This code is correct and ok, but lines 33-43 and 47-57 differ by just one word - the boolean. I wonder if you could create a reusable component that would accept the boolean and use it instead.

Reply:
I literally could not get this to work. My linter was conflicting with typescript and I couldn't find a solution.